### PR TITLE
fix(mcp): load workspace MCP config for native slash commands

### LINKED ISSUE #416: Approved project MCP servers are not loaded in new sessions
## Summary

Approved/configured project MCP servers are not always loaded or visible in newly created sessions.

## Observed behavior

In a project that has an approved MCP configuration file at `.aethon/mcp.toml`, creating a new session and running `/mcp` can show:

```text
MCP servers

No MCP servers are currently visible to Aethon.

Run /mcp setup to create or approve project MCP config.
```

This is confusing because the file tree shows the project MCP config exists:

```text
.aethon/
  mcp.toml
  startup.toml
```

The expected MCP servers were already configured/approved for the project, but the new session behaves as if no MCP config is visible or approved.

## Expected behavior

- New sessions for a project/workspace should load the approved MCP servers from that project's `.aethon/mcp.toml` automatically.
- `/mcp` in a new session should list the same approved servers that are visible in existing sessions for the same workspace.
- The user should not need to rerun `/mcp setup` just because they opened a new session.

## Why this matters

MCP availability affects tool use and agent capabilities. If a new session does not inherit approved project MCP servers, the agent may silently lose tools that are required for the task, and the UI suggests setup is missing even though the project is already configured.

## Suggested investigation

- Check whether project MCP approval/loading is scoped to an existing tab/session instead of the project/workspace.
- Verify that new tab/session creation sends the correct cwd/workspace root before MCP discovery runs.
- Confirm whether worker bridges load project-local `.aethon/mcp.toml` on startup and after tab creation.
- Compare MCP visibility between an existing session and a newly created session in the same workspace.
- Add a regression test: configure/approve project MCPs, create a new session for the same workspace, run/list MCP status, and assert the approved servers are present.

### PR DESCRIPTION
**Pull request by jamesbrink**
## Summary
- Resolve `/mcp` project roots from the invoking tab cwd or active workspace cwd instead of falling back to the main project path.
- Forward that cwd through native slash command dispatch so newly created sessions/workspaces load the expected approved `.aethon/mcp.toml` / project MCP configuration.
- Apply the cwd override when executing extension slash commands even if the pi session already exists, so reused sessions refresh MCP visibility for the correct workspace.
- Add frontend and bridge regressions for tab cwd, active workspace fallback, and existing-session native slash command behavior.

Closes #416.

## Validation
- `codex review --uncommitted` (clean after addressing two valid cwd edge-case findings)
- `bunx vitest run` (344 files / 2856 tests)
- `bunx tsc -b --noEmit`
- `bunx eslint .`

Note: Vitest still emits existing jsdom/xterm/component-registration warnings during the full suite, but all tests passed.

### CODE REVIEW

File: agent/nativeSlash.ts [hunk 1]
```diff
@@ -129,16 +129,28 @@ function tryRunExtensionSlashCommand(
...
+  if (cwdOverride && runner && typeof previousCwd === "string") {
+    runner.cwd = cwdOverride;
+  }
+  return Promise.resolve(runExtensionCommand.call(tab.session, text)).finally(
+    () => {
+      if (cwdOverride && runner && typeof previousCwd === "string") {
+        runner.cwd = previousCwd;
+      }
+    },
+  );
```
[the remark this answers is not in the document]
Comment (jamesbrink) [reply agent/nativeSlash.ts:157]: Addressed — the temporary cwd override now applies whenever the extension runner exists, even if it did not already expose a cwd field. The previous cwd is restored when present, and the temporary field is removed otherwise. Added a regression covering a runner with no preexisting cwd.
